### PR TITLE
Refactor node_network_configuration_policy resource

### PR DESCRIPTION
**Pull request by Anatw**
- Refactor _ports_backup to include both IP families.
 - Add teardown_absent_ifaces argument (default: True).
 - Add IP family global variables.

### CODE REVIEW

File: ocp_resources/node_network_configuration_policy.py [hunk 1]
```diff
@@ -254,13 +257,16 @@ def _ports_backup(self, ip_family):
         for port in self.ports:
             _port = self._get_port_from_nns(port_name=port)
             if _port:
-                self.ipv4_ports_backup_dict[port] = _port[ip_family]
+                if ip_family == IPV4_STR:
+                    self.ipv4_ports_backup_dict[port] = _port[ip_family]
+                else:  # ip_family == IPV6_STR
```
Comment (myakove) [inline ocp_resources/node_network_configuration_policy.py:262]: Please do `elif  ip_family == IPV6_STR`

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/node_network_configuration_policy.py (hunk 1 above)
Comment (Anatw) [reply ocp_resources/node_network_configuration_policy.py:262]: Ack.

File: ocp_resources/node_network_configuration_policy.py [hunk 2]
```diff
@@ -254,13 +257,16 @@ def _ports_backup(self, ip_family):
         for port in self.ports:
             _port = self._get_port_from_nns(port_name=port)
             if _port:
-                self.ipv4_ports_backup_dict[port] = _port[ip_family]
+                if ip_family == IPV4_STR:
+                    self.ipv4_ports_backup_dict[port] = _port[ip_family]
+                elif ip_family == IPV6_STR:
+                    self.ipv6_ports_backup_dict[port] = _port[ip_family]
```
Comment (myakove) [inline ocp_resources/node_network_configuration_policy.py:263]: Please raise if `ip_family` is not `IPV4_STR` or `IPV6_STR`

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/node_network_configuration_policy.py (hunk 2 above)
Comment (Anatw) [reply ocp_resources/node_network_configuration_policy.py:263]: ack

File: ocp_resources/node_network_configuration_policy.py [hunk 3]
```diff
@@ -254,13 +257,20 @@ def _ports_backup(self, ip_family):
...
+                else:
+                    raise TypeError(
```
Comment (rnetser) [inline ocp_resources/node_network_configuration_policy.py:265]: Should raise `ValueError`
(see https://docs.python.org/3/library/exceptions.html#TypeError)

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/node_network_configuration_policy.py (hunk 3 above)
Comment (Anatw) [reply ocp_resources/node_network_configuration_policy.py:265]: Ack.

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Pull request merged**